src/controller/proxy/local.go

```diff
@@ -171,7 +197,34 @@ func (l *localHelper) PushManifestList(ctx context.Context, repo string, ref str
 		return err
 	}
 	log.Debugf("The manifest list payload: %v", string(pl))
-	return l.PushManifest(repo, ref, newMan)
+	dig := digest.FromBytes(pl)
+	// Because the manifest list maybe updated, need to recheck if it is exist in local
+	inLocal, err := l.SameArtifact(ctx, repo, tag, string(dig))
+	if err != nil {
+		return err
+	}
+	if inLocal {
+		l.SendPullEvent(ctx, repo, tag)
```

Comment:
I don't think you should send a pull event in push func.

Reply:
Removed